tests/test_json_schema.py

```diff
@@ -2186,6 +2186,10 @@ class Example(BaseModel):
 
 def test_path_modify_schema():
     class MyPath(Path):
+        @classmethod
+        def __get_pydantic_core_schema__(cls, _source_type: Any, handler: GetCoreSchemaHandler) -> CoreSchema:
+            return handler(Path)
```

Comment:
Some of these are _trivial_, I'm happy to ask users to write this to avoid doing the wrong thing.
But also as per below, it would be relatively easy to provide some sort of opt-in functionality for these "simple" cases.

Reply:
Might be a good idea to add a link to custom types docs in the error message that gets raised if something is not a recognized type